editor-components/header/header32/header32.tsx

```diff
@@ -22,111 +25,391 @@ class Header32 extends BaseHeader {
+          value: [
+            {
+              type: "image",
+              key: "backgroundImage",
+              displayer: "Background Image",
+              value:
+                "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/66a383fd2f8a5b002ce6aae3?alt=media",
+            },
+            {
+              type: "string",
+              key: "imageTitle",
+              displayer: "Title",
+              value: "Architecture is our passion, design is our art",
+            },
+            {
+              type: "string",
+              key: "imageDescription",
+              displayer: "Description",
+              value:
+                "Architects offer design and planning services for buildings, landscapes, and interiors.",
+            },
+            {
+              type: "string",
+              key: "urlTitle",
+              displayer: "URL Title",
+              value: "Details",
+            },
+            {
+              type: "page",
+              key: "url",
+              displayer: "URL",
+              value: "",
+            },
+          ],
         },
       ],
-    }) 
+    });
+
+    this.addProp({
+      type: "boolean",
+      key: "animation",
+      displayer: "Enable Animation",
+      value: true,
+    });
+
+    this.addProp({
+      type: "boolean",
+      key: "enableOverlay",
+      displayer: "Enable Overlay",
+      value: true,
+    });
+
+    this.addProp({
+      type: "boolean",
+      key: "enableBackgroundImageOverlay",
+      displayer: "Enable Background Image Overlay",
+      value: true,
+    });
+
+    this.addProp({
+      type: "icon",
+      key: "leftButtonIcon",
+      displayer: "Previous Button",
+      value: "GrPrevious",
+    });
+
+    this.addProp({
+      type: "icon",
+      key: "rightButtonIcon",
+      displayer: "Next Button",
+      value: "GrNext",
+    });
+    this.setComponentState("activeSlide", 0);
+    this.setComponentState("slider-ref", React.createRef());
   }
 
   getName(): string {
     return "Header-32";
   }
+  
 
   render() {
-
+    const animation: boolean = this.getPropValue("animation");
+    const itemsCount = this.castToObject<Card[]>("header").length;
+  
     const settings = {
+      arrows: false,
       dots: true,
-      infinite: true,
-      speed: 500,
-      autoplay: false,
+      infinite: itemsCount > 1,
+      speed: 1500,
+      autoplay: true,
+      fade: animation,
       autoplaySpeed: 3000,
       slidesToShow: 1,
       slidesToScroll: 1,
+      beforeChange: (current: number, next: number) => {
+        if (this.getComponentState("activeSlide") !== next) {
+          this.setComponentState("activeSlide", next);
+        }
+      },
+      dotsClass: this.decorateCSS("dots"),
+      customPaging: function (i: number) {
+        return (
+          <div>
+            <button></button>
+            <span className={styles.dotIndex}>{`0${i + 1}`}</span>
+          </div>
```

Comment:
it looks like the button and span seem unnecessary. also, use decorateCss in classNames